src/cursors/row.rs

```diff
@@ -36,16 +45,47 @@ impl<T> RowCursor<T> {
     where
         T: Deserialize<'b>,
     {
+        let should_validate = match self.validation_mode {
+            ValidationMode::Disabled => false,
+            ValidationMode::Each => true,
+            ValidationMode::First(n) => self.rows_emitted < (n as u64),
+        };
+
         loop {
-            let mut slice = super::workaround_51132(self.bytes.slice());
+            if self.bytes.remaining() > 0 {
+                let mut slice = super::workaround_51132(self.bytes.slice());
+                let deserialize_result = if should_validate {
```

Comment:
```
if should_validate { ... } else { ... }
```

seems to significantly affect performance on this hot path, but it's not clear how to avoid using it if we want to preserve the validation mode logic